src/data.js

```diff
@@ -195,3 +195,54 @@ Role.prototype.json = function(resolver) {
 
 // Export Role
 exports.Role = Role;
+
+/**
+ * AtomicRoles represents the entire role configuration in one row, to support
+ * atomic updates.
+ */
+var AtomicRoles = Entity.configure({
```

Comment:
Oh, I thought it was 1MB :(

Reply:
Suggestions on how to solve this?  I don't think switching to postgres is a good project to tack onto the end of this one.  Given those options, I'd prefer to keep the existing storage -- while it's theoretically racy, in practice roles are only occasionally updated, and it would be easy to set up the ScopeResolver to keep using the old resolver whenever a new resolver contains a cycle.

Reply:
Okay, it's actually 256kb, let's see how close we are to that...

Reply:
From discussion, the race is in fact a vulnerability -- someone could maliciously create a role.  And refusing to adopt a cyclical resolver means that when the dynos restart daily, they will fail to start -- no good.  So options are
 * keep with azure tables if the current size is substantially smaller than [256k](https://github.com/taskcluster/azure-entities/blob/master/src/entitytypes.js#L489)
 * azure blobs (which do have sufficient semantics)

Reply:
The current set of roles are 245k.  So I'd say, no, we don't have room.